--- ../src_base/minecraft/src/net/minecraft/src/ChunkTFC.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/src/net/minecraft/src/ChunkTFC.java	0000-00-00 00:00:00.000000000 -0000
@@ -0,0 +1,65 @@
+package net.minecraft.src;
+
+public class ChunkTFC extends Chunk
+{
+
+    public ChunkTFC(World world, byte[] ids, byte[] metadata, byte[] ids2, byte[] metadata2, int chunkX, int chunkZ)
+    {
+        super(world, chunkX, chunkZ);
+        int height = ids.length / 256;
+
+        for (int x = 0; x < 16; ++x)
+        {
+            for (int z = 0; z < 16; ++z)
+            {
+                for (int y = 0; y < height; ++y)
+                {
+                    int index = x << 11 | z << 7 | y;
+                    int id    = ids[index] & 0xFF;
+                    int meta  = metadata[index] & 0x0F;
+
+                    if (id != 0)
+                    {
+                        int chunkY = (y >> 4);
+
+                        if (getBlockStorageArray()[chunkY] == null)
+                        {
+                            getBlockStorageArray()[chunkY] = new ExtendedBlockStorage(chunkY << 4);
+                        }
+
+                        getBlockStorageArray()[chunkY].setExtBlockID(x, y & 15, z, id);
+                        getBlockStorageArray()[chunkY].setExtBlockMetadata(x, y & 15, z, meta);
+                    }
+                }
+            }
+        }
+        
+        for (int x = 0; x < 16; ++x)
+        {
+            for (int z = 0; z < 16; ++z)
+            {
+                for (int y = 0; y < height; ++y)
+                {
+                    int index = x << 11 | z << 7 | y;
+                    int id    = ids2[index] & 0xFF;
+                    int meta  = metadata2[index] & 0x0F;
+
+                    if (id != 0)
+                    {
+                        int chunkY = 8+(y >> 4);
+
+                        if (getBlockStorageArray()[chunkY] == null)
+                        {
+                            int ind = (chunkY << 4);
+                            getBlockStorageArray()[chunkY] = new ExtendedBlockStorage(ind);
+                        }
+
+                        getBlockStorageArray()[chunkY].setExtBlockID(x, y & 15, z, id);
+                        getBlockStorageArray()[chunkY].setExtBlockMetadata(x, y & 15, z, meta);
+                    }
+                }
+            }
+        }
+    }
+
+}
