--- ../src_base/minecraft_server/net/minecraft/src/BlockLadder.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft_server/net/minecraft/src/BlockLadder.java	0000-00-00 00:00:00.000000000 -0000
@@ -72,9 +72,9 @@
     public boolean canPlaceBlockAt(World par1World, int par2, int par3, int par4)
     {
         return par1World.isBlockSolidOnSide(par2 - 1, par3, par4, 5) ||
-               par1World.isBlockSolidOnSide(par2 + 1, par3, par4, 4) ||
-               par1World.isBlockSolidOnSide(par2, par3, par4 - 1, 3) ||
-               par1World.isBlockSolidOnSide(par2, par3, par4 + 1, 2);
+                par1World.isBlockSolidOnSide(par2 + 1, par3, par4, 4) ||
+                par1World.isBlockSolidOnSide(par2, par3, par4 - 1, 3) ||
+                par1World.isBlockSolidOnSide(par2, par3, par4 + 1, 2);
     }
 
     /**
@@ -153,10 +153,10 @@
     {
         return 1;
     }
-    
+
     @Override
-    public boolean isLadder(World world, int x, int y, int z) 
+    public boolean isLadder(World world, int x, int y, int z)
     {
-    	return true;
+        return true;
     }
 }
