--- ../src_base/minecraft_server/net/minecraft/src/BlockTrapDoor.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft_server/net/minecraft/src/BlockTrapDoor.java	0000-00-00 00:00:00.000000000 -0000
@@ -4,7 +4,7 @@
 {
     /** Set this to allow trapdoors to remain free-floating */
     public static boolean disableValidation = false;
-    
+
     protected BlockTrapDoor(int par1, Material par2Material)
     {
         super(par1, par2Material);
@@ -242,7 +242,7 @@
         {
             return true;
         }
-        
+
         if (par5 == 0)
         {
             return false;
