--- ../src_base/minecraft_server/net/minecraft/src/EntitySquid.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft_server/net/minecraft/src/EntitySquid.java	0000-00-00 00:00:00.000000000 -0000
@@ -2,12 +2,12 @@
 
 public class EntitySquid extends EntityWaterMob
 {
-    public float field_21063_a = 0.0F;
-    public float field_21062_b = 0.0F;
-    public float field_21061_c = 0.0F;
-    public float field_21059_f = 0.0F;
-    public float field_21060_ak = 0.0F;
-    public float field_21058_al = 0.0F;
+    public float field_21089_a = 0.0F;
+    public float field_21088_b = 0.0F;
+    public float field_21087_c = 0.0F;
+    public float field_21086_f = 0.0F;
+    public float field_21085_g = 0.0F;
+    public float field_21084_h = 0.0F;
 
     /** angle of the tentacles in radians */
     public float tentacleAngle = 0.0F;
@@ -15,8 +15,8 @@
     /** the last calculated angle of the tentacles in radians */
     public float lastTentacleAngle = 0.0F;
     private float randomMotionSpeed = 0.0F;
-    private float field_21054_ap = 0.0F;
-    private float field_21053_aq = 0.0F;
+    private float field_21080_l = 0.0F;
+    private float field_21079_m = 0.0F;
     private float randomMotionVecX = 0.0F;
     private float randomMotionVecY = 0.0F;
     private float randomMotionVecZ = 0.0F;
@@ -26,7 +26,7 @@
         super(par1World);
         this.texture = "/mob/squid.png";
         this.setSize(0.95F, 0.95F);
-        this.field_21054_ap = 1.0F / (this.rand.nextFloat() + 1.0F) * 0.2F;
+        this.field_21080_l = 1.0F / (this.rand.nextFloat() + 1.0F) * 0.2F;
     }
 
     public int getMaxHealth()
@@ -127,19 +127,19 @@
     public void onLivingUpdate()
     {
         super.onLivingUpdate();
-        this.field_21062_b = this.field_21063_a;
-        this.field_21059_f = this.field_21061_c;
-        this.field_21058_al = this.field_21060_ak;
+        this.field_21088_b = this.field_21089_a;
+        this.field_21086_f = this.field_21087_c;
+        this.field_21084_h = this.field_21085_g;
         this.lastTentacleAngle = this.tentacleAngle;
-        this.field_21060_ak += this.field_21054_ap;
+        this.field_21085_g += this.field_21080_l;
 
-        if (this.field_21060_ak > ((float)Math.PI * 2F))
+        if (this.field_21085_g > ((float)Math.PI * 2F))
         {
-            this.field_21060_ak -= ((float)Math.PI * 2F);
+            this.field_21085_g -= ((float)Math.PI * 2F);
 
             if (this.rand.nextInt(10) == 0)
             {
-                this.field_21054_ap = 1.0F / (this.rand.nextFloat() + 1.0F) * 0.2F;
+                this.field_21080_l = 1.0F / (this.rand.nextFloat() + 1.0F) * 0.2F;
             }
         }
 
@@ -147,26 +147,26 @@
         {
             float var1;
 
-            if (this.field_21060_ak < (float)Math.PI)
+            if (this.field_21085_g < (float)Math.PI)
             {
-                var1 = this.field_21060_ak / (float)Math.PI;
+                var1 = this.field_21085_g / (float)Math.PI;
                 this.tentacleAngle = MathHelper.sin(var1 * var1 * (float)Math.PI) * (float)Math.PI * 0.25F;
 
                 if ((double)var1 > 0.75D)
                 {
                     this.randomMotionSpeed = 1.0F;
-                    this.field_21053_aq = 1.0F;
+                    this.field_21079_m = 1.0F;
                 }
                 else
                 {
-                    this.field_21053_aq *= 0.8F;
+                    this.field_21079_m *= 0.8F;
                 }
             }
             else
             {
                 this.tentacleAngle = 0.0F;
                 this.randomMotionSpeed *= 0.9F;
-                this.field_21053_aq *= 0.99F;
+                this.field_21079_m *= 0.99F;
             }
 
             if (!this.worldObj.isRemote)
@@ -179,12 +179,12 @@
             var1 = MathHelper.sqrt_double(this.motionX * this.motionX + this.motionZ * this.motionZ);
             this.renderYawOffset += (-((float)Math.atan2(this.motionX, this.motionZ)) * 180.0F / (float)Math.PI - this.renderYawOffset) * 0.1F;
             this.rotationYaw = this.renderYawOffset;
-            this.field_21061_c += (float)Math.PI * this.field_21053_aq * 1.5F;
-            this.field_21063_a += (-((float)Math.atan2((double)var1, this.motionY)) * 180.0F / (float)Math.PI - this.field_21063_a) * 0.1F;
+            this.field_21087_c += (float)Math.PI * this.field_21079_m * 1.5F;
+            this.field_21089_a += (-((float)Math.atan2((double)var1, this.motionY)) * 180.0F / (float)Math.PI - this.field_21089_a) * 0.1F;
         }
         else
         {
-            this.tentacleAngle = MathHelper.abs(MathHelper.sin(this.field_21060_ak)) * (float)Math.PI * 0.25F;
+            this.tentacleAngle = MathHelper.abs(MathHelper.sin(this.field_21085_g)) * (float)Math.PI * 0.25F;
 
             if (!this.worldObj.isRemote)
             {
@@ -194,7 +194,7 @@
                 this.motionZ = 0.0D;
             }
 
-            this.field_21063_a = (float)((double)this.field_21063_a + (double)(-90.0F - this.field_21063_a) * 0.02D);
+            this.field_21089_a = (float)((double)this.field_21089_a + (double)(-90.0F - this.field_21089_a) * 0.02D);
         }
     }
 
