--- ../src_base/minecraft_server/net/minecraft/src/EntityItem.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft_server/net/minecraft/src/EntityItem.java	0000-00-00 00:00:00.000000000 -0000
@@ -179,24 +179,27 @@
         if (!this.worldObj.isRemote)
         {
             int var2 = this.item.stackSize;
-            
+
             if (delayBeforeCanPickup == 0 && !ForgeHooks.onItemPickup(par1EntityPlayer, this))
             {
-                FMLServerHandler.instance().notifyItemPickup(this,par1EntityPlayer);
+                FMLServerHandler.instance().notifyItemPickup(this, par1EntityPlayer);
                 this.worldObj.playSoundAtEntity(this, "random.pop", 0.2F, ((rand.nextFloat() - rand.nextFloat()) * 0.7F + 1.0F) * 2.0F);
                 par1EntityPlayer.onItemPickup(this, var2);
+
                 if (item.stackSize <= 0)
                 {
                     setDead();
                 }
+
                 return;
             }
-            
+
             var2 = item.stackSize;
 
             if (this.delayBeforeCanPickup == 0 && par1EntityPlayer.inventory.addItemStackToInventory(this.item))
             {
-                FMLServerHandler.instance().notifyItemPickup(this,par1EntityPlayer);
+                FMLServerHandler.instance().notifyItemPickup(this, par1EntityPlayer);
+
                 if (this.item.itemID == Block.wood.blockID)
                 {
                     par1EntityPlayer.triggerAchievement(AchievementList.mineWood);
