--- ../src_base/minecraft_server/net/minecraft/src/IMinecraftRegistry.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft_server/net/minecraft/src/IMinecraftRegistry.java	0000-00-00 00:00:00.000000000 -0000
@@ -2,7 +2,6 @@
 
 public interface IMinecraftRegistry
 {
-
     public abstract void removeSpawn(String entityName, EnumCreatureType spawnList, BiomeGenBase... biomes);
 
     public abstract void removeSpawn(Class <? extends EntityLiving > entityClass, EnumCreatureType typeOfCreature, BiomeGenBase... biomes);
@@ -30,5 +29,4 @@
     public abstract void addShapelessRecipe(ItemStack output, Object... params);
 
     public abstract void addRecipe(ItemStack output, Object... params);
-
 }
\ No newline at end of file
