--- ../src_base/minecraft_server/net/minecraft/src/MLProp.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft_server/net/minecraft/src/MLProp.java	0000-00-00 00:00:00.000000000 -0000
@@ -28,21 +28,24 @@
     /**
      * Adds additional help to top of configuration file.
      */
+
 String info() default "";
 
     /**
      * Maximum value allowed if field is a number.
      */
+
 double max() default 1D;
 
     /**
      * Minimum value allowed if field is a number.
      */
+
 double min() default -1D;
 
     /**
      * Overrides the field name for property key.
      */
-String name() default "";
 
+String name() default "";
 }
