--- ../src_base/minecraft_server/net/minecraft/src/ModLoader.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft_server/net/minecraft/src/ModLoader.java	0000-00-00 00:00:00.000000000 -0000
@@ -606,13 +606,14 @@
     {
         return FMLCommonHandler.instance().isChannelActive(channel, player);
     }
-    
+
     /**
      * Stubbed method on the server to return a unique model id
-     * 
+     *
      */
     @Deprecated
-    public static int getUniqueBlockModelID(BaseMod mod, boolean flag) {
+    public static int getUniqueBlockModelID(BaseMod mod, boolean flag)
+    {
         return 0;
     }
 }
\ No newline at end of file
