--- ../src_base/minecraft_server/net/minecraft/src/ServerConfigurationManager.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft_server/net/minecraft/src/ServerConfigurationManager.java	0000-00-00 00:00:00.000000000 -0000
@@ -109,6 +109,7 @@
         {
             ((WorldServer)world).playerManager.removePlayer(par1EntityPlayerMP);
         }
+
         this.getPlayerManager(par1EntityPlayerMP.dimension).addPlayer(par1EntityPlayerMP);
         WorldServer var2 = this.mcServer.getWorldManager(par1EntityPlayerMP.dimension);
         var2.chunkProviderServer.loadChunk((int)par1EntityPlayerMP.posX >> 4, (int)par1EntityPlayerMP.posZ >> 4);
@@ -306,7 +307,6 @@
         par1EntityPlayerMP.playerNetServerHandler.sendPacket(new Packet9Respawn(par1EntityPlayerMP.dimension, (byte)par1EntityPlayerMP.worldObj.difficultySetting, var5.getWorldInfo().getTerrainType(), var5.getHeight(), par1EntityPlayerMP.itemInWorldManager.getGameType()));
         var4.removePlayer(par1EntityPlayerMP);
         par1EntityPlayerMP.isDead = false;
-
         WorldProvider pOld = DimensionManager.getProvider(var3);
         WorldProvider pNew = DimensionManager.getProvider(par2);
         double moveFactor = pOld.getMovementFactor() / pNew.getMovementFactor();
@@ -343,7 +343,6 @@
         par1EntityPlayerMP.itemInWorldManager.setWorld(var5);
         this.updateTimeAndWeather(par1EntityPlayerMP, var5);
         this.func_30008_g(par1EntityPlayerMP);
-        
         FMLServerHandler.instance().announceDimensionChange(par1EntityPlayerMP);
     }
 
