--- ../src_base/minecraft_server/net/minecraft/src/WorldProvider.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft_server/net/minecraft/src/WorldProvider.java	0000-00-00 00:00:00.000000000 -0000
@@ -39,6 +39,25 @@
         this.terrainType = par1World.getWorldInfo().getTerrainType();
         this.registerWorldChunkManager();
         this.generateLightBrightnessTable();
+
+        /**
+         * TFC Code: This is used to set the initial rock layers for each biome
+         * */
+        for (int i = 0; i < BiomeGenBase.biomeList.length; i++)
+        {
+            if (BiomeGenBase.biomeList[i] != null && !(BiomeGenBase.biomeList[i] instanceof BiomeGenHillsEdgeTFC) && !(BiomeGenBase.biomeList[i] instanceof BiomeGenRiverTFC))
+            {
+                BiomeGenBase.biomeList[i].SetupStone(par1World);
+            }
+            else if (BiomeGenBase.biomeList[i] instanceof BiomeGenHillsEdgeTFC)
+            {
+                BiomeGenBase.biomeList[i].copyBiomeRocks(BiomeGenBase.biomeList[i].biomeName.replace(" Edge", "").toLowerCase());
+            }
+            else if (BiomeGenBase.biomeList[i] instanceof BiomeGenRiverTFC)
+            {
+                BiomeGenBase.biomeList[i].copyBiomeRocks(BiomeGenBase.biomeList[i].biomeName.replace("River ", "").toLowerCase());
+            }
+        }
     }
 
     /**
@@ -84,7 +103,11 @@
     public boolean canCoordinateBeSpawn(int par1, int par2)
     {
         int var3 = this.worldObj.getFirstUncoveredBlock(par1, par2);
-        return var3 == Block.grass.blockID;
+        //return var3 == Block.grass.blockID;
+        /*
+         * This makes it so that the player can spawn on TFC's custom grass
+         */
+        return var3 == mod_TFC_Core.terraGrass.blockID || var3 == mod_TFC_Core.terraGrass2.blockID;
     }
 
     /**
@@ -141,30 +164,30 @@
     {
         return this.terrainType == WorldType.FLAT ? 4 : 64;
     }
-    
+
     /**
      * Returns the sub-folder of the world folder that this WorldProvider saves to.
      * EXA: DIM1, DIM-1
      * @return The sub-folder name to save this world's chunks to.
      */
     public abstract String getSaveFolder();
-    
+
     /**
      * A message to display to the user when they transfer to this dimension.
-     * 
+     *
      * @return The message to be displayed
      */
     public abstract String getWelcomeMessage();
-    
+
     /**
      * A Message to display to the user when they transfer out of this dismension.
-     * 
+     *
      * @return The message to be displayed
      */
     public abstract String getDepartMessage();
-    
+
     /**
-     * The dimensions movement factor. Relative to normal overworld. 
+     * The dimensions movement factor. Relative to normal overworld.
      * It is applied to the players position when they transfer dimensions.
      * Exa: Nether movement is 8.0
      * @return The movement factor
