--- ../src_base/minecraft_server/net/minecraft/src/WorldServer.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft_server/net/minecraft/src/WorldServer.java	0000-00-00 00:00:00.000000000 -0000
@@ -30,6 +30,7 @@
         {
             this.entityInstanceIdMap = new IntHashMap();
         }
+
         DimensionManager.setWorld(par4, this);
         playerManager = new PlayerManager(mcServer, par4, mcServer.propertyManagerObj.getIntProperty("view-distance", 10));
         entityTracker = new EntityTracker(mcServer, par4);
@@ -78,20 +79,23 @@
     public List getTileEntityList(int par1, int par2, int par3, int par4, int par5, int par6)
     {
         ArrayList var7 = new ArrayList();
-        for(int x = (par1 >> 4); x <= (par4 >> 4); x++)
+
+        for (int x = (par1 >> 4); x <= (par4 >> 4); x++)
         {
-            for(int z = (par3 >> 4); z <= (par6 >> 4); z++) 
+            for (int z = (par3 >> 4); z <= (par6 >> 4); z++)
             {
                 Chunk chunk = getChunkFromChunkCoords(x, z);
+
                 if (chunk != null)
                 {
-                    for(Object obj : chunk.chunkTileEntityMap.values())
+                    for (Object obj : chunk.chunkTileEntityMap.values())
                     {
                         TileEntity entity = (TileEntity)obj;
+
                         if (!entity.isInvalid())
                         {
-                            if (entity.xCoord >= par1 && entity.yCoord >= par2 && entity.zCoord >= par3 && 
-                                entity.xCoord <= par4 && entity.yCoord <= par5 && entity.zCoord <= par6)
+                            if (entity.xCoord >= par1 && entity.yCoord >= par2 && entity.zCoord >= par3 &&
+                                    entity.xCoord <= par4 && entity.yCoord <= par5 && entity.zCoord <= par6)
                             {
                                 var7.add(entity);
                             }
