--- ../src_base/minecraft_server/net/minecraft/src/WorldProvider.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft_server/net/minecraft/src/WorldProvider.java	0000-00-00 00:00:00.000000000 -0000
@@ -1,5 +1,8 @@
 package net.minecraft.src;
 
+import java.util.Random;
+
+import TFC.Core.TFC_Core;
 import net.minecraft.src.forge.DimensionManager;
 
 public abstract class WorldProvider
@@ -39,6 +42,10 @@
         this.terrainType = par1World.getWorldInfo().getTerrainType();
         this.registerWorldChunkManager();
         this.generateLightBrightnessTable();
+        /**
+         * TFC Code: This is used to set the initial rock layers for each biome
+         * */
+        TFC_Core.SetupWorld(par1World);
     }
 
     /**
@@ -77,7 +84,11 @@
     public boolean canCoordinateBeSpawn(int par1, int par2)
     {
         int var3 = this.worldObj.getFirstUncoveredBlock(par1, par2);
-        return var3 == Block.grass.blockID;
+        //return var3 == Block.grass.blockID;
+        /*
+         * This makes it so that the player can spawn on TFC's custom grass
+         */
+        return var3 == mod_TFC.terraGrass.blockID || var3 == mod_TFC.terraGrass2.blockID;
     }
 
     /**
