--- ../src_base/minecraft/net/minecraft/src/forge/IArmorTextureProvider.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/forge/IArmorTextureProvider.java	0000-00-00 00:00:00.000000000 -0000
@@ -14,13 +14,10 @@
  */
 public interface IArmorTextureProvider
 {
-
     /**
      * This interface has to return the path to a file that is the same
      * format as iron_1.png (or any of the other armor files). It will be
      * applied to the player skin when the armor is worn.
      */
     public String getArmorTextureFile(ItemStack itemstack);
-
 }
-
