--- ../src_base/minecraft/net/minecraft/src/forge/IGuiHandler.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/forge/IGuiHandler.java	0000-00-00 00:00:00.000000000 -0000
@@ -6,7 +6,7 @@
 public interface IGuiHandler
 {
     /**
-     * Returns a Container to be displayed to the user. 
+     * Returns a Container to be displayed to the user.
      * On the client side, this needs to return a instance of GuiScreen
      * On the server side, this needs to return a instance of Container
      *
