--- ../src_base/minecraft/net/minecraft/src/forge/IHighlightHandler.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/forge/IHighlightHandler.java	0000-00-00 00:00:00.000000000 -0000
@@ -19,4 +19,3 @@
      */
     public boolean onBlockHighlight(RenderGlobal render, EntityPlayer player, MovingObjectPosition target, int i, ItemStack currentItem, float partialTicks);
 }
-
