--- ../src_base/minecraft/net/minecraft/src/forge/IMinecartCollisionHandler.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/forge/IMinecartCollisionHandler.java	0000-00-00 00:00:00.000000000 -0000
@@ -12,7 +12,6 @@
  */
 public interface IMinecartCollisionHandler
 {
-
     /**
      * This basically replaces the function of the same name in EnityMinecart.
      * Code in IMinecartHooks.applyEntityCollisionHook is still run.
@@ -47,4 +46,3 @@
      */
     public AxisAlignedBB getBoundingBox(EntityMinecart cart);
 }
-
