--- ../src_base/minecraft/net/minecraft/src/forge/ISaveEventHandler.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/forge/ISaveEventHandler.java	0000-00-00 00:00:00.000000000 -0000
@@ -11,7 +11,7 @@
      * @param world The world being loaded.
      */
     public void onWorldLoad(World world);
-    
+
     /**
      * Called whenever the world is saving. Use this to save extra data alongside the world, eg. maps.
      * @param world The world being saved.
