--- ../src_base/minecraft/net/minecraft/src/forge/ISoundHandler.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/forge/ISoundHandler.java	0000-00-00 00:00:00.000000000 -0000
@@ -3,19 +3,19 @@
 import net.minecraft.src.SoundManager;
 import net.minecraft.src.SoundPoolEntry;
 
-public interface ISoundHandler 
+public interface ISoundHandler
 {
     /**
-     * This event is raised by the SoundManager when it does its first setup of the 
+     * This event is raised by the SoundManager when it does its first setup of the
      * SoundSystemConfig's codecs, use this function to add your own codecs.
      * @param soundManager The SoundManager instance
      */
     void onSetupAudio(SoundManager soundManager);
 
     /**
-     * Raised by the SoundManager.loadSoundSettings, this would be a good place for 
+     * Raised by the SoundManager.loadSoundSettings, this would be a good place for
      * adding your custom sounds to the SoundPool.
-     * 
+     *
      * @param soundManager The SoundManager instance
      */
     void onLoadSoundSettings(SoundManager soundManager);
@@ -24,9 +24,9 @@
      * Raised when the SoundManager tries to play a Background Music file,
      * If you return null from this function it will prevent the sound from being played,
      * you can return a different entry if you want to change the sound being played.
-     * 
+     *
      * If you do not want to change anything, just return the passed in entry.
-     * 
+     *
      * @param soundManager The SoundManager instance
      * @param entry The current entry that will be played
      * @return The new sound entry to play, or the current one passed in.
@@ -36,12 +36,12 @@
     /**
      * Raised when the SoundManager tries to play a 'Streaming' file,
      * in vanilla it is only the Jukebox that uses this function.
-     * 
+     *
      * If you return null from this function it will prevent the sound from being played,
      * you can return a different entry if you want to change the sound being played.
-     * 
+     *
      * If you do not want to change anything, just return the passed in entry.
-     * 
+     *
      * @param soundManager The SoundManager instance
      * @param entry The current entry that will be played
      * @param soundName The name of the request sound
@@ -55,12 +55,12 @@
     /***
      * Raised when the SoundManager tries to play a normal sound,
      * dogs barking, footsteps, etc. THe majority of all sounds during normal game play.
-     * 
+     *
      * If you return null from this function it will prevent the sound from being played,
      * you can return a different entry if you want to change the sound being played.
-     * 
-     * If you do not want to change anything, just return the passed in entry. 
-     * 
+     *
+     * If you do not want to change anything, just return the passed in entry.
+     *
      * @param soundManager The SoundManager instance
      * @param entry The current entry that will be played
      * @param soundName The name of the request sound
@@ -77,12 +77,12 @@
      * Raised when the SoundManager tries to play a effect sound,
      * currently the only known sounds are 'random.click' when a GUI button is clicked,
      * or 'portal.trigger' and 'portal.travel' when the player is near/inside a portal.
-     * 
+     *
      * If you return null from this function it will prevent the sound from being played,
      * you can return a different entry if you want to change the sound being played.
-     * 
-     * If you do not want to change anything, just return the passed in entry. 
-     * 
+     *
+     * If you do not want to change anything, just return the passed in entry.
+     *
      * @param soundManager The SoundManager instance
      * @param entry The current entry that will be played
      * @param soundName The name of the request sound
@@ -91,5 +91,4 @@
      * @return The new sound entry to play, or the current one passed in.
      */
     SoundPoolEntry onPlaySoundEffect(SoundManager soundManager, SoundPoolEntry entry, String soundName, float volume, float pitch);
-
 }
