--- ../src_base/minecraft/net/minecraft/src/forge/ITextureLoadHandler.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/forge/ITextureLoadHandler.java	0000-00-00 00:00:00.000000000 -0000
@@ -1,6 +1,6 @@
 package net.minecraft.src.forge;
 
-public interface ITextureLoadHandler 
+public interface ITextureLoadHandler
 {
     public void onTextureLoad(String textureName, int textureID);
 }
