--- ../src_base/minecraft/net/minecraft/src/forge/MinecraftForgeClient.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/forge/MinecraftForgeClient.java	0000-00-00 00:00:00.000000000 -0000
@@ -37,7 +37,7 @@
     {
         ForgeHooksClient.registerRenderContextHandler(texture, subid, handler);
     }
-    
+
     /**
      * Registers a Texture Load Handler
      * @param handler The handler
@@ -46,7 +46,7 @@
     {
         ForgeHooksClient.textureLoadHandlers.add(handler);
     }
-    
+
     /**
      * Registers a Render Last Handler
      * @param handler The handler
@@ -55,7 +55,7 @@
     {
         ForgeHooksClient.renderWorldLastHandlers.add(handler);
     }
-    
+
     /**
      * Registers a Sound Handler
      * @param handler The handler
@@ -114,7 +114,7 @@
     {
         return ForgeHooksClient.renderPass;
     }
-    
+
     private static IItemRenderer[] customItemRenderers = new IItemRenderer[Item.itemsList.length];
 
     /** Register a custom renderer for a specific item. This can be used to
@@ -132,10 +132,12 @@
     public static IItemRenderer getItemRenderer(ItemStack item, ItemRenderType type)
     {
         IItemRenderer renderer = customItemRenderers[item.itemID];
-        if (renderer != null && renderer.handleRenderType(item, type)) 
+
+        if (renderer != null && renderer.handleRenderType(item, type))
         {
             return customItemRenderers[item.itemID];
         }
+
         return null;
     }
 
@@ -146,6 +148,7 @@
         {
             return;
         }
+
         hasInit = true;
         ForgeHooks.setPacketHandler(new PacketHandlerClient());
     }
@@ -154,12 +157,12 @@
     {
         init();
     }
-    
+
     /***
      * This is a function that is used to enforce deprecation of code.
      * It checks the current Display's title against the passed in argument.
      * If they do not match (such is the case in different versionf of MC) it exits the process with a error
-     * 
+     *
      * @param version The version to find, usually "Minecraft Minecraft 1.2.3"
      * @param message The error message to display in the crash log
      */
