--- ../src_base/minecraft/net/minecraft/src/forge/NetworkMod.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/forge/NetworkMod.java	0000-00-00 00:00:00.000000000 -0000
@@ -29,18 +29,18 @@
     {
         return false;
     }
-    
+
     /**
      * Called when the 'small' data packet is received for this NetworkMod,
-     * Internally, this utilizes the Packet131MapData packet. This is useful 
-     * data that is sent rapidly and would like to save the overhead of the 
+     * Internally, this utilizes the Packet131MapData packet. This is useful
+     * data that is sent rapidly and would like to save the overhead of the
      * 250 custom payload packet.
-     * 
+     *
      * Data is limited to 256 bytes.
-     * 
+     *
      * @param net
      * @param id
      * @param data
      */
-    public void onPacketData(NetworkManager net, short id, byte[] data){}
+    public void onPacketData(NetworkManager net, short id, byte[] data) {}
 }
