--- ../src_base/minecraft/net/minecraft/src/forge/Property.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/forge/Property.java	0000-00-00 00:00:00.000000000 -0000
@@ -10,23 +10,23 @@
     public String name;
     public String value;
     public String comment;
-    
+
     /**
      * Returns the value in this property as a integer,
      * if the value is not a valid integer, it will return -1.
-     * 
+     *
      * @return The value
      */
     public int getInt()
     {
         return getInt(-1);
     }
-    
+
     /**
      * Returns the value in this property as a integer,
      * if the value is not a valid integer, it will return the
      * provided default.
-     * 
+     *
      * @param _default The default to provide if the current value is not a valid integer
      * @return The value
      */
@@ -41,7 +41,7 @@
             return _default;
         }
     }
-    
+
     /**
      * Checks if the current value stored in this property can be converted to an integer.
      * @return True if the vslue can be converted to an integer
@@ -58,12 +58,12 @@
             return false;
         }
     }
-    
+
     /**
      * Returns the value in this property as a boolean,
      * if the value is not a valid boolean, it will return the
      * provided default.
-     * 
+     *
      * @param _default The default to provide
      * @return The value as a boolean, or the default
      */
@@ -78,7 +78,7 @@
             return _default;
         }
     }
-    
+
     /**
      * Checks if the current value held by this property is a valid boolean value.
      * @return True if it is a boolean value
