--- ../src_base/minecraft_server/net/minecraft/server/MinecraftServer.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft_server/net/minecraft/server/MinecraftServer.java	0000-00-00 00:00:00.000000000 -0000
@@ -133,7 +133,7 @@
     public long[] field_48082_x = new long[100];
     private RConThreadQuery rconQueryThread;
     private RConThreadMain rconMainThread;
-    
+
     public int spawnProtectionSize = 16;
 
     public MinecraftServer()
@@ -274,6 +274,7 @@
             logger.info("Converting map!");
             par1ISaveFormat.convertMapFormat(par2Str, new ConvertProgressUpdater(this));
         }
+
         /*
         this.worldMngr = new WorldServer[3];
         this.field_40028_g = new long[this.worldMngr.length][100];
@@ -284,8 +285,8 @@
         boolean var7 = this.propertyManagerObj.getBooleanProperty("generate-structures", true);
         WorldSettings var8 = new WorldSettings(par3, var6, var7, false, par5WorldType);
         AnvilSaveHandler var9 = new AnvilSaveHandler(new File("."), par2Str, true);
-        
         WorldServer overWorld = new WorldServer(this, var9, par2Str, 0, var8);
+
         for (Integer id : DimensionManager.getIDs())
         {
             WorldServer world = (id == 0 ? overWorld : new WorldServerMulti(this, var9, par2Str, id, var8, overWorld));
@@ -295,8 +296,8 @@
             world.getWorldInfo().setGameType(var6);
             worldTickTimes.put(id, new long[100]);
         }
-        configManager.setPlayerManager(new WorldServer[]{ overWorld });
 
+        configManager.setPlayerManager(new WorldServer[] { overWorld });
         short var22 = 196;
         long var23 = System.currentTimeMillis();
 
@@ -504,7 +505,6 @@
     private void doTick()
     {
         FMLServerHandler.instance().onPreTick();
-        
         long var1 = System.nanoTime();
         ArrayList var3 = new ArrayList();
         Iterator var4 = field_6037_b.keySet().iterator();
@@ -594,7 +594,6 @@
         this.field_48076_G = Packet.field_48101_l;
         this.field_48082_x[this.deathTime % 100] = Packet.field_48102_m - this.field_48077_H;
         this.field_48077_H = Packet.field_48102_m;
-
         FMLServerHandler.instance().onPostTick();
     }
 
