--- ../src_base/minecraft_server/cpw/mods/fml/server/ModLoaderModContainer.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft_server/cpw/mods/fml/server/ModLoaderModContainer.java	0000-00-00 00:00:00.000000000 -0000
@@ -206,22 +206,27 @@
             {
                 n = Double.parseDouble(val);
             }
+
             if (Float.class.isAssignableFrom(type))
             {
                 n = Float.parseFloat(val);
             }
+
             if (Long.class.isAssignableFrom(type))
             {
                 n = Long.parseLong(val);
             }
+
             if (Integer.class.isAssignableFrom(type))
             {
                 n = Integer.parseInt(val);
             }
+
             if (Short.class.isAssignableFrom(type))
             {
                 n = Short.parseShort(val);
             }
+
             if (Byte.class.isAssignableFrom(type))
             {
                 n = Byte.parseByte(val);
@@ -315,7 +320,7 @@
     }
 
     /**
-     * Find all the BaseMods in the system 
+     * Find all the BaseMods in the system
      * @return
      */
     public static List<BaseMod> findAll()
@@ -429,14 +434,13 @@
 
             if ("required-before".equals(depparts[0]) || "before".equals(depparts[0]))
             {
-            	postDependencies.add(depparts[1]);
+                postDependencies.add(depparts[1]);
             }
 
             if ("required-after".equals(depparts[0]) || "after".equals(depparts[0]))
             {
                 preDependencies.add(depparts[1]);
             }
-
         }
     }
 
@@ -469,10 +473,10 @@
         {
             computeDependencies();
         }
+
         return preDependencies;
     }
 
-
     public String toString()
     {
         return modSource.getName();
