--- ../src_base/common/net/minecraft/src/BiomeGenBase.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/common/net/minecraft/src/BiomeGenBase.java	0000-00-00 00:00:00.000000000 -0000
@@ -97,7 +97,7 @@
     /**
      * Is true (default) if the biome support rain (desert and nether can't have rain)
      */
-    private boolean enableRain;
+    protected boolean enableRain;
 
     /** The id number to this biome, and its index in the biomeList array. */
     public final int biomeID;
@@ -294,7 +294,7 @@
     /**
      * Gets an integer representation of this biome's rainfall
      */
-    public final int getIntRainfall()
+    public int getIntRainfall()
     {
         return (int)(this.rainfall * 65536.0F);
     }
@@ -302,7 +302,7 @@
     /**
      * Gets an integer representation of this biome's temperature
      */
-    public final int getIntTemperature()
+    public int getIntTemperature()
     {
         return (int)(this.temperature * 65536.0F);
     }
@@ -312,7 +312,7 @@
     /**
      * Gets a floating point representation of this biome's rainfall
      */
-    public final float getFloatRainfall()
+    public float getFloatRainfall()
     {
         return this.rainfall;
     }
@@ -320,7 +320,7 @@
     /**
      * Gets a floating point representation of this biome's temperature
      */
-    public final float getFloatTemperature()
+    public float getFloatTemperature()
     {
         return this.temperature;
     }
