--- ../src_base/common/net/minecraft/src/BlockFlowing.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/common/net/minecraft/src/BlockFlowing.java	0000-00-00 00:00:00.000000000 -0000
@@ -50,7 +50,7 @@
         int var6 = this.getFlowDecay(par1World, par2, par3, par4);
         byte var7 = 1;
 
-        if (this.blockMaterial == Material.lava && !par1World.provider.isHellWorld)
+        if (this.blockMaterial == Material.lava /*&& !par1World.provider.isHellWorld*/)
         {
             var7 = 2;
         }
