--- ../src_base/common/net/minecraft/src/EntityMinecart.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/common/net/minecraft/src/EntityMinecart.java	0000-00-00 00:00:00.000000000 -0000
@@ -61,7 +61,7 @@
     public EntityMinecart(World par1World)
     {
         super(par1World);
-        this.cargoItems = new ItemStack[36];
+        this.cargoItems = new ItemStack[18];
         this.fuel = 0;
         this.field_70499_f = false;
         this.preventEntitySpawning = true;
@@ -882,7 +882,7 @@
      */
     public int getSizeInventory()
     {
-        return (minecartType == 1 && getClass() == EntityMinecart.class ? 27 : 0);
+        return (minecartType == 1 && getClass() == EntityMinecart.class ? 18 : 0);
     }
 
     /**
