--- ../src_base/common/net/minecraft/src/ItemMonsterPlacer.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/common/net/minecraft/src/ItemMonsterPlacer.java	0000-00-00 00:00:00.000000000 -0000
@@ -1,5 +1,6 @@
 package net.minecraft.src;
 
+import TFC.Core.TFCSettings;
 import cpw.mods.fml.common.Side;
 import cpw.mods.fml.common.asm.SideOnly;
 import java.util.Iterator;
@@ -89,6 +90,10 @@
                     return true;
                 }
 
+                if (var8 instanceof TFC.Entities.EntityAnimalTFC)
+                {
+                	((TFC.Entities.EntityAnimalTFC) var8).setGrowingAge (-TFCSettings.dayLength * ((TFC.Entities.EntityAnimalTFC)var8).adultAge);
+                }
                 par0World.spawnEntityInWorld(var8);
                 ((EntityLiving)var8).playLivingSound();
             }
