--- ../src_base/common/net/minecraft/src/WorldType.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/common/net/minecraft/src/WorldType.java	0000-00-00 00:00:00.000000000 -0000
@@ -161,12 +161,12 @@
 
     public int getMinimumSpawnHeight(World world)
     {
-        return this == FLAT ? 4 : 64;
+        return this == FLAT ? 4 : 145;
     }
 
     public double getHorizon(World world)
     {
-        return this == FLAT ? 0.0D : 63.0D;
+        return this == FLAT ? 0.0D : 144.0D;
     }
 
     public boolean hasVoidParticles(boolean var1)
