--- ../src_base/common/net/minecraft/src/WorldProvider.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/common/net/minecraft/src/WorldProvider.java	0000-00-00 00:00:00.000000000 -0000
@@ -1,5 +1,7 @@
 package net.minecraft.src;
 
+import TFC.Core.TFCSettings;
+import TFC.Core.TFC_Core;
 import net.minecraftforge.client.SkyProvider;
 import net.minecraftforge.common.DimensionManager;
 import cpw.mods.fml.common.Side;
@@ -44,6 +46,10 @@
         this.terrainType = par1World.getWorldInfo().getTerrainType();
         this.registerWorldChunkManager();
         this.generateLightBrightnessTable();
+        /**
+         * TFC Code: This is used to set the initial rock layers for each biome
+         * */
+        TFC_Core.SetupWorld(par1World);
     }
 
     /**
@@ -82,7 +88,8 @@
     public boolean canCoordinateBeSpawn(int par1, int par2)
     {
         int var3 = this.worldObj.getFirstUncoveredBlock(par1, par2);
-        return var3 == Block.grass.blockID;
+        //return var3 == Block.grass.blockID;
+        return var3 == TFCBlocks.terraGrass.blockID || var3 == TFCBlocks.terraGrass2.blockID;
     }
 
     /**
@@ -90,8 +97,8 @@
      */
     public float calculateCelestialAngle(long par1, float par3)
     {
-        int var4 = (int)(par1 % 24000L);
-        float var5 = ((float)var4 + par3) / 24000.0F - 0.25F;
+        int var4 = (int)(par1 % (long)TFCSettings.dayLength);
+        float var5 = ((float)var4 + par3) / (float)TFCSettings.dayLength - 0.25F;
 
         if (var5 < 0.0F)
         {
@@ -112,7 +119,7 @@
     @SideOnly(Side.CLIENT)
     public int getMoonPhase(long par1, float par3)
     {
-        return (int)(par1 / 24000L) % 8;
+        return (int)(par1 / (long)TFCSettings.dayLength) % 8;
     }
 
     /**
@@ -199,7 +206,7 @@
      */
     public float getCloudHeight()
     {
-        return 128.0F;
+        return 256.0F;
     }
 
     @SideOnly(Side.CLIENT)
