--- ../src_base/minecraft_server/cpw/mods/fml/common/FMLCommonHandler.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft_server/cpw/mods/fml/common/FMLCommonHandler.java	0000-00-00 00:00:00.000000000 -0000
@@ -23,14 +23,14 @@
 
 /**
  * The main class for non-obfuscated hook handling code
- * 
- * Anything that doesn't require obfuscated or client/server specific code should 
+ *
+ * Anything that doesn't require obfuscated or client/server specific code should
  * go in this handler
- * 
+ *
  * It also contains a reference to the sided handler instance that is valid
  * allowing for common code to access specific properties from the obfuscated world
  * without a direct dependency
- * 
+ *
  * @author cpw
  *
  */
@@ -53,11 +53,10 @@
      */
     private Map<Object, Set<String>> activeChannels = new HashMap<Object, Set<String>>();
     /**
-     * The delegate for side specific data and functions 
+     * The delegate for side specific data and functions
      */
     private IFMLSidedHandler sidedDelegate;
 
-
     /**
      * We register our delegate here
      * @param handler
@@ -68,7 +67,7 @@
     }
 
     /**
-     * Pre-tick the mods 
+     * Pre-tick the mods
      */
     public void gameTickStart()
     {
@@ -249,8 +248,9 @@
     {
         return sidedDelegate.loadBaseModMod(clazz, canonicalFile);
     }
-    
-    public File getMinecraftRootDirectory() {
+
+    public File getMinecraftRootDirectory()
+    {
         return sidedDelegate.getMinecraftRootDirectory();
     }
 }
