--- ../src_base/minecraft_server/cpw/mods/fml/common/ICraftingHandler.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft_server/cpw/mods/fml/common/ICraftingHandler.java	0000-00-00 00:00:00.000000000 -0000
@@ -15,16 +15,15 @@
 
 /**
  * Return a crafting handler for the mod container to call
- * 
+ *
  * @author cpw
  *
  */
 public interface ICraftingHandler
 {
-
     /**
      * The object array contains these three arguments
-     * 
+     *
      * @param player
      * @param craftedItem
      * @param craftingGrid
