--- ../src_base/minecraft_server/cpw/mods/fml/common/IWorldGenerator.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft_server/cpw/mods/fml/common/IWorldGenerator.java	0000-00-00 00:00:00.000000000 -0000
@@ -16,9 +16,9 @@
 import java.util.Random;
 
 /**
- * This is called back during world generation. The {@link #generate(Random, int, int, Object...)} method passes in some additional data that 
+ * This is called back during world generation. The {@link #generate(Random, int, int, Object...)} method passes in some additional data that
  * can be cast into minecraft objects.
- * 
+ *
  * @author cpw
  *
  */
@@ -26,14 +26,14 @@
 {
     /**
      * Generate some world
-     * 
+     *
      * @param random the chunk specific {@link Random} as built in {@link FMLServerHandler#onChunkPopulate(net.minecraft.src.IChunkProvider, int, int, net.minecraft.src.World, net.minecraft.src.IChunkProvider)}.
      * @param chunkX the block X coordinate of this chunk.
      * @param chunkZ the block Z coordinate of this chunk.
      * @param world : additionalData[0] The minecraft {@link net.minecraft.src.World} we're generating for.
      * @param generator : additionalData[1] The {@link net.minecraft.src.IChunkProvider} that is generating.
      * @param chunkProvider : additionalData[2] {@link net.minecraft.src.IChunkProvider} that is requesting the world generation.
-     * 
+     *
      */
     public void generate(Random random, int chunkX, int chunkZ, Object...additionalData);
 }
