--- ../src_base/minecraft/net/minecraft/src/forge/packets/PacketModList.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/forge/packets/PacketModList.java	0000-00-00 00:00:00.000000000 -0000
@@ -25,6 +25,7 @@
         if (!isServer)
         {
             data.writeInt(Mods.length);
+
             for (String mod : Mods)
             {
                 data.writeUTF(mod);
@@ -33,6 +34,7 @@
         else
         {
             data.writeInt(ModIDs.size());
+
             for (Entry<Integer, String> entry : ModIDs.entrySet())
             {
                 data.writeInt(entry.getKey());
@@ -47,9 +49,11 @@
         if (isServer)
         {
             Length = data.readInt();
+
             if (Length >= 0)
             {
                 Mods = new String[Length];
+
                 for (int x = 0; x < Length; x++)
                 {
                     Mods[x] = data.readUTF();
@@ -59,6 +63,7 @@
         else
         {
             Length = data.readInt();
+
             for (int x = 0; x < Length; x++)
             {
                 ModIDs.put(data.readInt(), data.readUTF());
@@ -79,6 +84,7 @@
         {
             StringBuilder ret = new StringBuilder();
             ret.append(toString()).append('\n');
+
             if (Mods != null)
             {
                 for (String mod : Mods)
@@ -93,6 +99,7 @@
                     ret.append(String.format("    %03d ", mod.getKey()) + mod.getValue() + '\n');
                 }
             }
+
             return ret.toString();
         }
         else
@@ -100,5 +107,4 @@
             return toString();
         }
     }
-
 }
