--- ../src_base/minecraft/net/minecraft/src/forge/packets/PacketEntitySpawn.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/forge/packets/PacketEntitySpawn.java	0000-00-00 00:00:00.000000000 -0000
@@ -31,18 +31,15 @@
     public Object metadata;
     private ISpawnHandler handler;
 
-    public PacketEntitySpawn(){}
+    public PacketEntitySpawn() {}
     public PacketEntitySpawn(Entity ent, NetworkMod mod, int type)
     {
         entityID = ent.entityId;
-
         posX = MathHelper.floor_double(ent.posX * 32D);
         posY = MathHelper.floor_double(ent.posY * 32D);
         posZ = MathHelper.floor_double(ent.posZ * 32D);
-
         typeID = type;
         modID = MinecraftForge.getModID(mod);
-
         yaw      = (byte)(ent.rotationYaw * 256.0F / 360.0F);
         pitch    = (byte)(ent.rotationPitch * 256.0F / 360.0F);
         yawHead  = (byte)(ent instanceof EntityLiving ? ((EntityLiving)ent).rotationYawHead * 256.0F / 360.0F : 0);
@@ -56,16 +53,42 @@
             double mX = ent.motionX;
             double mY = ent.motionY;
             double mZ = ent.motionZ;
-            if (mX < -maxVel) mX = -maxVel;
-            if (mY < -maxVel) mY = -maxVel;
-            if (mZ < -maxVel) mZ = -maxVel;
-            if (mX >  maxVel) mX =  maxVel;
-            if (mY >  maxVel) mY =  maxVel;
-            if (mZ >  maxVel) mZ =  maxVel;
+
+            if (mX < -maxVel)
+            {
+                mX = -maxVel;
+            }
+
+            if (mY < -maxVel)
+            {
+                mY = -maxVel;
+            }
+
+            if (mZ < -maxVel)
+            {
+                mZ = -maxVel;
+            }
+
+            if (mX >  maxVel)
+            {
+                mX =  maxVel;
+            }
+
+            if (mY >  maxVel)
+            {
+                mY =  maxVel;
+            }
+
+            if (mZ >  maxVel)
+            {
+                mZ =  maxVel;
+            }
+
             speedX = (int)(mX * 8000D);
             speedY = (int)(mY * 8000D);
             speedZ = (int)(mZ * 8000D);
         }
+
         if (ent instanceof ISpawnHandler)
         {
             handler = (ISpawnHandler)ent;
@@ -84,12 +107,14 @@
         data.writeByte(yawHead);
         ((DataWatcher)metadata).writeWatchableObjects(data);
         data.writeInt(throwerID);
+
         if (throwerID != 0)
         {
             data.writeShort(speedX);
             data.writeShort(speedY);
             data.writeShort(speedZ);
         }
+
         if (handler != null)
         {
             handler.writeSpawnData(data);
@@ -109,6 +134,7 @@
         yawHead   = data.readByte();
         metadata  = DataWatcher.readWatchableObjects(data);
         throwerID = data.readInt();
+
         if (throwerID != 0)
         {
             speedX = data.readShort();
