--- ../src_base/minecraft/net/minecraft/src/BiomeGenBase.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/BiomeGenBase.java	0000-00-00 00:00:00.000000000 -0000
@@ -8,7 +8,7 @@
 public abstract class BiomeGenBase
 {
     /** An array of all the biomes, indexed by biome id. */
-    public static final BiomeGenBase[] biomeList = new BiomeGenBase[256];
+    public static BiomeGenBase[] biomeList = new BiomeGenBase[256];
     public static final BiomeGenBase ocean = (new BiomeGenOcean(0)).setColor(112).setBiomeName("Ocean").setMinMaxHeight(-1.0F, 0.4F);
     public static final BiomeGenBase plains = (new BiomeGenPlains(1)).setColor(9286496).setBiomeName("Plains").setTemperatureRainfall(0.8F, 0.4F);
     public static final BiomeGenBase desert = (new BiomeGenDesert(2)).setColor(16421912).setBiomeName("Desert").setDisableRain().setTemperatureRainfall(2.0F, 0.0F).setMinMaxHeight(0.1F, 0.2F);
@@ -93,7 +93,7 @@
     /**
      * Is true (default) if the biome support rain (desert and nether can't have rain)
      */
-    private boolean enableRain;
+    protected boolean enableRain;
 
     /** The id number to this biome, and its index in the biomeList array. */
     public final int biomeID;
@@ -286,7 +286,7 @@
     /**
      * Gets an integer representation of this biome's rainfall
      */
-    public final int getIntRainfall()
+    public int getIntRainfall()
     {
         return (int)(this.rainfall * 65536.0F);
     }
@@ -294,7 +294,7 @@
     /**
      * Gets an integer representation of this biome's temperature
      */
-    public final int getIntTemperature()
+    public int getIntTemperature()
     {
         return (int)(this.temperature * 65536.0F);
     }
@@ -302,7 +302,7 @@
     /**
      * Gets a floating point representation of this biome's rainfall
      */
-    public final float getFloatRainfall()
+    public float getFloatRainfall()
     {
         return this.rainfall;
     }
@@ -310,7 +310,7 @@
     /**
      * Gets a floating point representation of this biome's temperature
      */
-    public final float getFloatTemperature()
+    public float getFloatTemperature()
     {
         return this.temperature;
     }
