--- ../src_base/minecraft/net/minecraft/src/BlockFire.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/BlockFire.java	0000-00-00 00:00:00.000000000 -0000
@@ -106,7 +106,7 @@
     {
         Block base = Block.blocksList[par1World.getBlockId(par2, par3 - 1, par4)];
         boolean var6 = (base != null && base.isFireSource(par1World, par2, par3 - 1, par4, par1World.getBlockMetadata(par2, par3 - 1, par4), 0));
-        
+
         if (par1World.worldProvider instanceof WorldProviderEnd && par1World.getBlockId(par2, par3 - 1, par4) == Block.bedrock.blockID)
         {
             var6 = true;
@@ -205,18 +205,19 @@
             }
         }
     }
-    
+
     /** Deprecated for a side-sensitive version */
     @Deprecated
     private void tryToCatchBlockOnFire(World par1World, int par2, int par3, int par4, int par5, Random par6Random, int par7)
     {
         tryToCatchBlockOnFire(par1World, par2, par3, par4, par5, par6Random, par7, 0);
     }
-    
+
     private void tryToCatchBlockOnFire(World par1World, int par2, int par3, int par4, int par5, Random par6Random, int par7, int face)
     {
         int var8 = 0;
         Block block = Block.blocksList[par1World.getBlockId(par2, par3, par4)];
+
         if (block != null)
         {
             var8 = block.getFlammability(par1World, par2, par3, par4, par1World.getBlockMetadata(par2, par3, par4), face);
@@ -255,11 +256,11 @@
     private boolean canNeighborBurn(World par1World, int par2, int par3, int par4)
     {
         return canBlockCatchFire(par1World, par2 + 1, par3, par4, 4) ||
-               canBlockCatchFire(par1World, par2 - 1, par3, par4, 5) ||
-               canBlockCatchFire(par1World, par2, par3 - 1, par4, 1) ||
-               canBlockCatchFire(par1World, par2, par3 + 1, par4, 0) ||
-               canBlockCatchFire(par1World, par2, par3, par4 - 1, 3) ||
-               canBlockCatchFire(par1World, par2, par3, par4 + 1, 2);
+                canBlockCatchFire(par1World, par2 - 1, par3, par4, 5) ||
+                canBlockCatchFire(par1World, par2, par3 - 1, par4, 1) ||
+                canBlockCatchFire(par1World, par2, par3 + 1, par4, 0) ||
+                canBlockCatchFire(par1World, par2, par3, par4 - 1, 3) ||
+                canBlockCatchFire(par1World, par2, par3, par4 + 1, 2);
     }
 
     /**
@@ -436,10 +437,10 @@
             }
         }
     }
-    
+
     /**
      * Side sensitive version that calls the block function.
-     * 
+     *
      * @param world The current world
      * @param x X Position
      * @param y Y Position
@@ -449,17 +450,19 @@
      */
     public boolean canBlockCatchFire(IBlockAccess world, int x, int y, int z, int face)
     {
-    	Block block = Block.blocksList[world.getBlockId(x, y, z)];
-    	if (block != null)
-    	{
-    		return block.isFlammable(world, x, y, z, world.getBlockMetadata(x, y, z), face);
-    	}
-    	return false;
+        Block block = Block.blocksList[world.getBlockId(x, y, z)];
+
+        if (block != null)
+        {
+            return block.isFlammable(world, x, y, z, world.getBlockMetadata(x, y, z), face);
+        }
+
+        return false;
     }
-    
+
     /**
      * Side sensitive version that calls the block function.
-     * 
+     *
      * @param world The current world
      * @param x X Position
      * @param y Y Position
@@ -470,12 +473,14 @@
      */
     public int getChanceToEncourageFire(World world, int x, int y, int z, int oldChance, int face)
     {
-    	int newChance = 0;
-    	Block block = Block.blocksList[world.getBlockId(x, y, z)];
-    	if (block != null)
-    	{
-    		newChance = block.getFireSpreadSpeed(world, x, y, z, world.getBlockMetadata(x, y, z), face);
-    	}
+        int newChance = 0;
+        Block block = Block.blocksList[world.getBlockId(x, y, z)];
+
+        if (block != null)
+        {
+            newChance = block.getFireSpreadSpeed(world, x, y, z, world.getBlockMetadata(x, y, z), face);
+        }
+
         return (newChance > oldChance ? newChance : oldChance);
-    }    
+    }
 }
