--- ../src_base/minecraft/net/minecraft/src/BlockTallGrass.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/BlockTallGrass.java	0000-00-00 00:00:00.000000000 -0000
@@ -72,32 +72,35 @@
     {
         super.harvestBlock(par1World, par2EntityPlayer, par3, par4, par5, par6);
     }
-    
+
     @Override
     public ArrayList<ItemStack> getBlockDropped(World world, int x, int y, int z, int meta, int fortune)
     {
         ArrayList<ItemStack> ret = new ArrayList<ItemStack>();
+
         if (world.rand.nextInt(8) != 0)
         {
             return ret;
         }
-        
+
         ItemStack item = ForgeHooks.getGrassSeed(world);
+
         if (item != null)
         {
             ret.add(item);
         }
+
         return ret;
     }
 
     @Override
-    public boolean isShearable(ItemStack item, World world, int x, int y, int z) 
+    public boolean isShearable(ItemStack item, World world, int x, int y, int z)
     {
         return true;
     }
 
     @Override
-    public ArrayList<ItemStack> onSheared(ItemStack item, World world, int x, int y, int z, int fortune) 
+    public ArrayList<ItemStack> onSheared(ItemStack item, World world, int x, int y, int z, int fortune)
     {
         ArrayList<ItemStack> ret = new ArrayList<ItemStack>();
         ret.add(new ItemStack(this, 1, world.getBlockMetadata(x, y, z)));
