--- ../src_base/minecraft/net/minecraft/src/BlockVine.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/BlockVine.java	0000-00-00 00:00:00.000000000 -0000
@@ -439,21 +439,21 @@
     }
 
     @Override
-    public boolean isShearable(ItemStack item, World world, int x, int y, int z) 
+    public boolean isShearable(ItemStack item, World world, int x, int y, int z)
     {
         return true;
     }
 
     @Override
-    public ArrayList<ItemStack> onSheared(ItemStack item, World world, int x, int y, int z, int fortune) 
+    public ArrayList<ItemStack> onSheared(ItemStack item, World world, int x, int y, int z, int fortune)
     {
         ArrayList<ItemStack> ret = new ArrayList<ItemStack>();
         ret.add(new ItemStack(this, 1, 0));
         return ret;
     }
-    
+
     @Override
-    public boolean isLadder(World world, int x, int y, int z) 
+    public boolean isLadder(World world, int x, int y, int z)
     {
         return true;
     }
