--- ../src_base/minecraft/net/minecraft/src/EntityPlayer.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/EntityPlayer.java	0000-00-00 00:00:00.000000000 -0000
@@ -213,6 +213,7 @@
             else
             {
                 itemInUse.getItem().onUsingItemTick(itemInUse, this, itemInUseCount);
+
                 if (this.itemInUseCount <= 25 && this.itemInUseCount % 4 == 0)
                 {
                     this.updateItemUse(var1, 5);
@@ -622,14 +623,17 @@
     public EntityItem dropOneItem()
     {
         ItemStack stack = inventory.getCurrentItem();
+
         if (stack == null)
         {
             return null;
         }
+
         if (stack.getItem().onDroppedByPlayer(stack, this))
         {
             return dropPlayerItemWithRandomChoice(inventory.decrStackSize(inventory.currentItem, 1), false);
         }
+
         return null;
     }
 
@@ -703,11 +707,11 @@
     {
         return getCurrentPlayerStrVsBlock(par1Block, 0);
     }
-    
+
     public float getCurrentPlayerStrVsBlock(Block par1Block, int meta)
     {
         ItemStack stack = inventory.getCurrentItem();
-        float var2 = (stack == null ? 1.0F : stack.getItem().getStrVsBlock(stack, par1Block, meta));        
+        float var2 = (stack == null ? 1.0F : stack.getItem().getStrVsBlock(stack, par1Block, meta));
         float var3 = var2;
         int var4 = EnchantmentHelper.getEfficiencyModifier(this.inventory);
 
@@ -1004,10 +1008,12 @@
         }
 
         par2 = ArmorProperties.ApplyArmor(this, inventory.armorInventory, par1DamageSource, par2);
+
         if (par2 <= 0)
         {
             return;
         }
+
         par2 = this.applyPotionDamageCalculations(par1DamageSource, par2);
         this.addExhaustion(par1DamageSource.getHungerDamage());
         this.health -= par2;
@@ -1042,6 +1048,7 @@
         {
             return;
         }
+
         if (!par1Entity.interact(this))
         {
             ItemStack var2 = this.getCurrentEquippedItem();
@@ -1107,11 +1114,14 @@
         {
             return;
         }
+
         ItemStack stack = getCurrentEquippedItem();
+
         if (stack != null && stack.getItem().onLeftClickEntity(stack, this, par1Entity))
         {
             return;
         }
+
         if (par1Entity.canAttackWithItem())
         {
             int var2 = this.inventory.getDamageVsEntity(par1Entity);
@@ -1255,11 +1265,12 @@
     public EnumStatus sleepInBedAt(int par1, int par2, int par3)
     {
         EnumStatus customSleep = ForgeHooks.sleepInBedAt(this, par1, par2, par3);
+
         if (customSleep != null)
         {
             return customSleep;
         }
-        
+
         if (!this.worldObj.isRemote)
         {
             if (this.isPlayerSleeping() || !this.isEntityAlive())
@@ -1742,6 +1753,7 @@
                     return 101;
                 }
             }
+
             var3 = par1ItemStack.getItem().getIconIndex(par1ItemStack, par2, this, itemInUse, itemInUseCount);
         }
 
@@ -1915,10 +1927,10 @@
     }
 
     public void func_50009_aI() {}
-    
+
     /**
-     * Opens a Gui for the player. 
-     * 
+     * Opens a Gui for the player.
+     *
      * @param mod The mod associated with the gui
      * @param ID The ID number for the Gui
      * @param world The World
@@ -1931,9 +1943,11 @@
         if (this instanceof EntityPlayerSP)
         {
             IGuiHandler handler = MinecraftForge.getGuiHandler(mod);
+
             if (handler != null)
             {
                 GuiScreen screen = (GuiScreen)handler.getGuiElement(ID, this, world, x, y, z);
+
                 if (screen != null)
                 {
                     ModLoader.getMinecraftInstance().displayGuiScreen(screen);
