--- ../src_base/minecraft/net/minecraft/src/EntitySquid.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/EntitySquid.java	0000-00-00 00:00:00.000000000 -0000
@@ -230,6 +230,6 @@
      */
     public boolean getCanSpawnHere()
     {
-        return this.posY > 45.0D && this.posY < 63.0D && super.getCanSpawnHere();
+        return this.posY > 128.0D && this.posY < 145.0D && super.getCanSpawnHere();
     }
 }
