--- ../src_base/minecraft/net/minecraft/src/EntityRenderer.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/EntityRenderer.java	0000-00-00 00:00:00.000000000 -0000
@@ -13,6 +13,8 @@
 import org.lwjgl.opengl.GLContext;
 import org.lwjgl.util.glu.GLU;
 
+import TFC.Core.TFCHeat;
+
 public class EntityRenderer
 {
     public static boolean anaglyphEnable = false;
@@ -1415,7 +1417,7 @@
                             double var35;
                             float var32;
 
-                            if (var42.getWorldChunkManager().getTemperatureAtHeight(var31, var26) >= 0.15F)
+                            if (TFCHeat.getNormalizedTemp(var42.getWorldChunkManager().getTemperatureAtHeight(var31, var26)) > 0)
                             {
                                 if (var18 != 0)
                                 {
