--- ../src_base/minecraft/net/minecraft/src/EntitySheep.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/EntitySheep.java	0000-00-00 00:00:00.000000000 -0000
@@ -266,21 +266,23 @@
     }
 
     @Override
-    public boolean isShearable(ItemStack item, World world, int X, int Y, int Z) 
+    public boolean isShearable(ItemStack item, World world, int X, int Y, int Z)
     {
         return !getSheared() && !isChild();
     }
 
     @Override
-    public ArrayList<ItemStack> onSheared(ItemStack item, World world, int X, int Y, int Z, int fortune) 
+    public ArrayList<ItemStack> onSheared(ItemStack item, World world, int X, int Y, int Z, int fortune)
     {
         ArrayList<ItemStack> ret = new ArrayList<ItemStack>();
         setSheared(true);
         int i = 1 + rand.nextInt(3);
+
         for (int j = 0; j < i; j++)
         {
             ret.add(new ItemStack(Block.cloth.blockID, 1, getFleeceColor()));
         }
+
         return ret;
     }
 }
