--- ../src_base/minecraft/net/minecraft/src/ItemHoe.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/ItemHoe.java	0000-00-00 00:00:00.000000000 -0000
@@ -28,6 +28,7 @@
                 par1ItemStack.damageItem(1, par2EntityPlayer);
                 return true;
             }
+
             int var8 = par3World.getBlockId(par4, par5, par6);
             int var9 = par3World.getBlockId(par4, par5 + 1, par6);
 
