--- ../src_base/minecraft/net/minecraft/src/PlayerControllerSP.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/PlayerControllerSP.java	0000-00-00 00:00:00.000000000 -0000
@@ -37,11 +37,12 @@
     public boolean onPlayerDestroyBlock(int par1, int par2, int par3, int par4)
     {
         ItemStack stack = mc.thePlayer.getCurrentEquippedItem();
+
         if (stack != null && stack.getItem().onBlockStartBreak(stack, par1, par2, par3, mc.thePlayer))
         {
             return false;
         }
-        
+
         int var5 = this.mc.theWorld.getBlockId(par1, par2, par3);
         int var6 = this.mc.theWorld.getBlockMetadata(par1, par2, par3);
         boolean var7 = super.onPlayerDestroyBlock(par1, par2, par3, par4);
@@ -202,33 +203,35 @@
      */
     public boolean onPlayerRightClick(EntityPlayer par1EntityPlayer, World par2World, ItemStack par3ItemStack, int par4, int par5, int par6, int par7)
     {
-        if (par3ItemStack != null && 
-            par3ItemStack.getItem() != null && 
-            par3ItemStack.getItem().onItemUseFirst(par3ItemStack, par1EntityPlayer, par2World, par4, par5, par6, par7))
+        if (par3ItemStack != null &&
+                par3ItemStack.getItem() != null &&
+                par3ItemStack.getItem().onItemUseFirst(par3ItemStack, par1EntityPlayer, par2World, par4, par5, par6, par7))
         {
-                return true;
+            return true;
         }
-        
+
         int var8 = par2World.getBlockId(par4, par5, par6);
+
         if (var8 > 0 && Block.blocksList[var8].blockActivated(par2World, par4, par5, par6, par1EntityPlayer))
         {
             return true;
         }
-        
+
         if (par3ItemStack == null)
         {
             return false;
         }
-        
+
         if (!par3ItemStack.useItem(par1EntityPlayer, par2World, par4, par5, par6, par7))
         {
             return false;
         }
-        
+
         if (par3ItemStack.stackSize <= 0)
         {
             ForgeHooks.onDestroyCurrentItem(par1EntityPlayer, par3ItemStack);
         }
+
         return true;
     }
 
