--- ../src_base/minecraft/net/minecraft/src/RailLogic.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/RailLogic.java	0000-00-00 00:00:00.000000000 -0000
@@ -29,12 +29,10 @@
         this.trackY = par4;
         this.trackZ = par5;
         int var6 = par2World.getBlockId(par3, par4, par5);
-        
         BlockRail target = (BlockRail)Block.blocksList[var6];
         int var7 = target.getBasicRailMetadata(par2World, null, par3, par4, par5);
         isPoweredRail = !target.isFlexibleRail(par2World, par3, par4, par5);
         canMakeSlopes = target.canMakeSlopes(par2World, par3, par4, par5);
-        
         this.setConnections(var7);
     }
 
