--- ../src_base/minecraft/net/minecraft/src/RegionFileCache.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/RegionFileCache.java	0000-00-00 00:00:00.000000000 -0000
@@ -50,6 +50,7 @@
     /**
      * Saves the current Chunk Map Cache
      */
+
     public static synchronized void clearRegionFileReferences()
     {
         Iterator var0 = regionsByFilename.values().iterator();
