--- ../src_base/minecraft/net/minecraft/src/RenderBiped.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/RenderBiped.java	0000-00-00 00:00:00.000000000 -0000
@@ -36,10 +36,9 @@
             this.modelBipedMain.bipedRightArm.postRender(0.0625F);
             GL11.glTranslatef(-0.0625F, 0.4375F, 0.0625F);
             float var4;
-
             IItemRenderer customRenderer = MinecraftForgeClient.getItemRenderer(var3, EQUIPPED);
             boolean is3D = (customRenderer != null && customRenderer.shouldUseRenderHelper(EQUIPPED, var3, BLOCK_3D));
-            
+
             if (var3.itemID < 256 && (is3D || RenderBlocks.renderItemIn3d(Block.blocksList[var3.itemID].getRenderType())))
             {
                 var4 = 0.5F;
