--- ../src_base/minecraft/net/minecraft/src/RenderSnowMan.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/RenderSnowMan.java	0000-00-00 00:00:00.000000000 -0000
@@ -28,10 +28,9 @@
         {
             GL11.glPushMatrix();
             this.snowmanModel.field_40305_c.postRender(0.0625F);
-
             IItemRenderer customRenderer = MinecraftForgeClient.getItemRenderer(var3, EQUIPPED);
             boolean is3D = (customRenderer != null && customRenderer.shouldUseRenderHelper(EQUIPPED, var3, BLOCK_3D));
-            
+
             if (is3D || RenderBlocks.renderItemIn3d(Block.blocksList[var3.itemID].getRenderType()))
             {
                 float var4 = 0.625F;
