--- ../src_base/minecraft/net/minecraft/src/RenderEngine.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/RenderEngine.java	0000-00-00 00:00:00.000000000 -0000
@@ -163,6 +163,7 @@
                 {
                     System.out.printf("Warning: Texture %s not preloaded, will cause render glitches!\n", par1Str);
                 }
+
                 this.singleIntBuffer.clear();
                 GLAllocation.generateTextureNames(this.singleIntBuffer);
                 int var6 = this.singleIntBuffer.get(0);
