--- ../src_base/minecraft/net/minecraft/src/SoundManager.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/SoundManager.java	0000-00-00 00:00:00.000000000 -0000
@@ -39,7 +39,7 @@
     /** RNG. */
     private Random rand = new Random();
     private int ticksBeforeMusic;
-    
+
     public static int MUSIC_INTERVAL = 12000;
 
     public SoundManager()
@@ -59,6 +59,7 @@
         {
             this.tryToSetLibraryAndCodecs();
         }
+
         ModCompatibilityClient.audioModLoad(this);
         ForgeHooksClient.onLoadSoundSettings(this);
     }
@@ -318,12 +319,12 @@
     {
         return soundPoolSounds;
     }
-    
+
     public SoundPool getStreamingPool()
     {
         return soundPoolStreaming;
     }
-    
+
     public SoundPool getMusicPool()
     {
         return soundPoolMusic;
