--- ../src_base/minecraft/net/minecraft/src/TileEntityRendererPiston.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/TileEntityRendererPiston.java	0000-00-00 00:00:00.000000000 -0000
@@ -31,6 +31,7 @@
             {
                 GL11.glShadeModel(GL11.GL_FLAT);
             }
+
             ForgeHooksClient.beforeBlockRender(var9, blockRenderer);
             var10.startDrawingQuads();
             var10.setTranslation((double)((float)par2 - (float)par1TileEntityPiston.xCoord + par1TileEntityPiston.getOffsetX(par8)), (double)((float)par4 - (float)par1TileEntityPiston.yCoord + par1TileEntityPiston.getOffsetY(par8)), (double)((float)par6 - (float)par1TileEntityPiston.zCoord + par1TileEntityPiston.getOffsetZ(par8)));
