--- ../src_base/minecraft/net/minecraft/src/WorldProvider.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/WorldProvider.java	0000-00-00 00:00:00.000000000 -0000
@@ -1,5 +1,8 @@
 package net.minecraft.src;
 
+import java.util.Random;
+
+import net.minecraft.src.TFC_Game.TFC_Game;
 import net.minecraft.src.forge.DimensionManager;
 
 public abstract class WorldProvider
@@ -39,6 +42,30 @@
         this.terrainType = par1World.getWorldInfo().getTerrainType();
         this.registerWorldChunkManager();
         this.generateLightBrightnessTable();
+        /**
+         * TFC Code: This is used to set the initial rock layers for each biome
+         * */
+        Random R = new Random(par1World.getSeed());
+        for(int i = 0; i < BiomeGenBase.biomeList.length; i++)
+        {
+        	if(BiomeGenBase.biomeList[i] != null && !(BiomeGenBase.biomeList[i] instanceof BiomeGenHillsEdgeTFC) && !(BiomeGenBase.biomeList[i] instanceof BiomeGenRiverTFC))
+        	{
+        		BiomeGenBase.biomeList[i].SetupStone(par1World, R);
+        	}
+        	else if(BiomeGenBase.biomeList[i] instanceof BiomeGenHillsEdgeTFC)
+        	{
+        		BiomeGenBase.biomeList[i].copyBiomeRocks(BiomeGenBase.biomeList[i].biomeName.replace(" Edge", "").toLowerCase());
+        	}
+        	else if(BiomeGenBase.biomeList[i] instanceof BiomeGenRiverTFC)
+        	{
+        		BiomeGenBase.biomeList[i].copyBiomeRocks(BiomeGenBase.biomeList[i].biomeName.replace("River ", "").toLowerCase());
+        	}
+        	if(BiomeGenBase.biomeList[i] != null)
+        	{
+        	    BiomeGenBase.biomeList[i].SetupTrees(par1World, R);
+        	}
+        }
+        TFC_Game.registerAnvilRecipes(R,par1World);
     }
 
     /**
@@ -77,7 +104,8 @@
     public boolean canCoordinateBeSpawn(int par1, int par2)
     {
         int var3 = this.worldObj.getFirstUncoveredBlock(par1, par2);
-        return var3 == Block.grass.blockID;
+        //return var3 == Block.grass.blockID;
+        return var3 == mod_TFC_Core.terraGrass.blockID || var3 == mod_TFC_Core.terraGrass2.blockID;
     }
 
     /**
@@ -184,7 +212,7 @@
      */
     public float getCloudHeight()
     {
-        return 128.0F;
+        return 256.0F;
     }
 
     public boolean isSkyColored()
@@ -202,7 +230,8 @@
 
     public int getAverageGroundLevel()
     {
-        return this.terrainType.getSeaLevel(this.worldObj);
+        return 145;
+        //return this.terrainType.getSeaLevel(this.worldObj);
     }
 
     /**
