--- ../src_base/minecraft/net/minecraft/src/WorldChunkManager.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/src/WorldChunkManager.java	0000-00-00 00:00:00.000000000 -0000
@@ -93,7 +93,10 @@
      */
     public float getTemperatureAtHeight(float par1, int par2)
     {
-        return par1;
+        float temp = par1;
+        if(par2 > 180)
+            temp -= temp * (par2-180)/90;
+        return temp;
     }
 
     /**
