--- ../src_base/minecraft/net/minecraft/server/MinecraftServer.java	0000-00-00 00:00:00.000000000 -0000
+++ ../src/minecraft/net/minecraft/server/MinecraftServer.java	0000-00-00 00:00:00.000000000 -0000
@@ -0,0 +1,19 @@
+package net.minecraft.server;
+
+import net.minecraft.src.WorldServer;
+
+public class MinecraftServer
+{
+
+    public WorldServer getWorldManager(int i)
+    {
+        // TODO Auto-generated method stub
+        return null;
+    }
+    
+    public static boolean isClient()
+    {
+        return true;
+    }
+
+}
